vk_video_decoder/libs/VulkanVideoFrameBuffer/VulkanVideoFrameBuffer.cpp

```diff
@@ -1063,6 +1112,20 @@ int32_t NvPerFrameDecodeImageSet::init(const VulkanDeviceContext* vkDevCtx,
             if (result != VK_SUCCESS) {
                 return -1;
             }
+            if (pDpbImageCreateInfo != nullptr) {
```

Comment:
I'm not sure why do we need all this code for re-indexing.
I also don't think we need the additional m_dpbImageView and all the complexity associated with this additional image.

Here is the idea about how the application is structured:

There are two images in NvPerFrameDecodeResources. m_frameImageView and m_displayImageView.

m_frameImageView: is a both OUT and DPB image for the COINCIDE (NV and intel) case. It is only DPB for the DISTINCT (AMD) case.

m_displayImageView is the decoder's OUT image for DISTINCT. It is also the NV out image when the output is required to be linear (non-optimal).  I would suspect AMD supports linear output for the OUT decoded images, correct?

So, I think, following this logic,  we should not need a third image.

At the decoder's side, when we are getting the out and DPB images, the configuration of the NvPerFrameDecodeResources should know what to return in NvPerFrameDecodeResources::GetImageSetNewLayout().

in the AMD's case it should return the separate images in pDpbPictureResource and pOutputPictureResource. For NV and Intel it should return the same image for both.

Does this make sense? I don't mind renaming m_frameImageView to  m_outDpbImageView and m_displayImageView to m_outImageView, BTW.